pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -81,26 +83,33 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	return
 }
 
-func (i *DefaultGatherer) GetPodsWithoutSidecar(kubeClient kubernetes.Interface, retryOpts []retry.Option) (podsList v1.PodList, err error) {
+func (i *DefaultGatherer) GetPodsWithoutSidecar(kubeClient kubernetes.Interface, retryOpts []retry.Option, sidecarInjectionEnabledbyDefault bool) (podsList v1.PodList, err error) {
 	allPodsWithNamespaceAnnotations, err := getAllPodsWithNamespaceAnnotations(kubeClient, retryOpts)
 	if err != nil {
 		return
 	}
 
 	// filter pods
-	podsList = getPodsWithAnnotation(allPodsWithNamespaceAnnotations)
+	podsList = getPodsWithAnnotation(allPodsWithNamespaceAnnotations, sidecarInjectionEnabledbyDefault)
 	podsList = getPodsWithoutSidecar(podsList)
 	return
 }
 
-func getPodsWithAnnotation(inputPodsList v1.PodList) (outputPodsList v1.PodList) {
+func getPodsWithAnnotation(inputPodsList v1.PodList, sidecarInjectionEnabledbyDefault bool) (outputPodsList v1.PodList) {
 	inputPodsList.DeepCopyInto(&outputPodsList)
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
 		if pod.Annotations["sidecar.istio.io/inject"] == "false" || pod.Annotations["reconciler/namespace-istio-injection"] == "disabled" {
```

Comment:
```suggestion
		if !sidecarInjectionEnabledbyDefault {
				_, namespaceLabeled := pod.Annotations["reconciler/namespace-istio-injection"]
				_, podAnnotated := pod.Annotations["sidecar.istio.io/inject"]
				if !namespaceLabeled && !podAnnotated {
					continue
				}     
		if pod.Annotations["sidecar.istio.io/inject"] == "false" || pod.Annotations["reconciler/namespace-istio-injection"] == "disabled" {
			continue
			}
```

Reply:
I think the logic you have added is what we need. My suggestion is to move it out of if-else and check it before checking disabled annotations.